File: PHP/MinimumNumberOfSecondsToMakeMountainHeightZeroTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MinimumNumberOfSecondsToMakeMountainHeightZeroTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MinimumNumberOfSecondsToMakeMountainHeightZero.php");

class MinimumNumberOfSecondsToMakeMountainHeightZeroTest extends TestCase {

    public function testMinimumNumberOfSwapsToMakeTheStringBalanced() {
        $obj = new MinimumNumberOfSecondsToMakeMountainHeightZero();
        //Test 1
        $want = 3;
        $got = $obj->minNumberOfSeconds(4, [2, 1, 1]);
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 12;
        $got = $obj->minNumberOfSeconds(10, [3, 2, 2, 4]);
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = 15;
        $got = $obj->minNumberOfSeconds(5, [1]);
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MinimumNumberOfSecondsToMakeMountainHeightZero_test.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMinimumNumberOfSecondsToMakeMountainHeightZero
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMinimumNumberOfSecondsToMakeMountainHeightZero(t *testing.T) {
	tables := []struct {
		want int64
		s    int
		lang []int
	}{
		{3, 4, []int{2, 1, 1}},
		{12, 10, []int{3, 2, 2, 4}},
		{15, 5, []int{1}},
	}

	for _, table := range tables {
		got := minNumberOfSeconds(table.s, table.lang)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
